c/Juliet_Test/Juliet_Test_Suite_v1.3_for_C_Cpp/C/testcases/preprocessing_script.sh

```diff
@@ -0,0 +1,92 @@
+#!/bin/bash
+
+# This file is part of the SV-Benchmarks collection of verification tasks:
+# https://github.com/sosy-lab/sv-benchmarks
+# SPDX-FileCopyrightText: 2020 The SV-Benchmarks Community
+# SPDX-License-Identifier: CC0-1.0
+
+# This script can only be called from sv-benchmarks/c/Juliet_Test/Juliet_Test_Suite_v1.3_for_C_Cpp/C/testcases
+# Running this script the .c files in each CWE directory are preprocessed and a _bad.i , _good.i and .yml file are created for each .c file.
+# The preprocessed/yml files are saved in the preprocessed folder in each CWE directory in testcases.
+# Files with a windows.h header can only be preprocessed on a windows system.
+
+
+# options for preprocessing
+OPTIONS="-E -P -D INCLUDEMAIN=1 -m64 -I "../testcasesupport""
+
+# property_file
+property_file=" "
+
+#for loop runs over all folders in testcases
+for folder in */
+do
+    # get the matching property_file for the CWE-Id
+    cwe_folder="$(basename $folder)" # get the plain directory name
+    cwe="${cwe_folder%${cwe_folder#??????}}" # extract the first 6 chars from the string
```

Comment:
**context:** In theory there is a nice `manifest.xml` along the original Juliet set which already contains machine-readable structured descriptive data about the test cases. Parsing this stuff from directory names is redundant. Code to determine the CWE from the folder name apparently already exists here: https://github.com/sosy-lab/sv-benchmarks/blob/064dfb14bd152626c6101623ed0a9ba66416ec56/c/Juliet_Test/Juliet_Test_Suite_v1.3_for_C_Cpp/C/create_per_cwe_files.py#L221

Granted that python script is not very readable and implementing stuff there will not improve the situation. I think this script here is useful and simple enough to keep it for the task it tries to achieve. No reason to overcomplicate things by making them more generic.